### PR TITLE
Return None when Trivia is missing for Ident in SynLongIdent.

### LINKED ISSUE #13842: `SynLongIdent.IdentsWithTrivia` throws exceptions
```fsharp
module Module

A.
```

[Ast](https://fsprojects.github.io/fantomas-tools/#/ast?data=N4KABGBEAmCmBmBLAdrAzpAXFSAacUiaAYmolmPAIYA2as%2BEkAxgPZwWQC27ArjbDABZPgIA6yCQEEAdJBABfIA). Trying to get the idents with trivia leads to the following exception:
```
Message = “difference between idents and trivia”

StackTraceString = “
  at FSharp.Compiler.Syntax.SynLongIdent.get_IdentsWithTrivia()
    in C:\Developer\fsharp-rebase\src\Compiler\SyntaxTree\SyntaxTree.fs:line 51
 at JetBrains.ReSharper.Plugins.FSharp.Psi.LanguageService.Parsing.FSharpExpressionTreeBuilder.ProcessLongIdentifierExpression(SynLongIdent lidWithDots, Range range)
    in C:\Developer\resharper-fsharp\ReSharper.FSharp\src\FSharp.Psi.Features\src\Parsing\FSharpImplTreeBuilder.fs:line 1188
```

I can make a workaround, but it's suspicious that the trivia is missing (i.e. empty list instead of `[None]`), and it should't probably throw instead of return `None` in this case.

cc @nojaf

### PR DESCRIPTION
**Pull request by nojaf**
Somewhat fixes #13842.

@auduchinok can you give some more examples where the trivia list is smaller than the ident list?
I would personally like to address all these cases when the SynLongIdent is created.

The partial fix I've added to `member this.IdentsWithTrivia` doesn't feel right.
These lists should in theory never be out of sync. Even when recovery is kicking in.

### CODE REVIEW

File: src/Compiler/SyntaxTree/SyntaxTree.fs [hunk 1]
```diff
@@ -47,10 +47,14 @@ type SynLongIdent =
...
-            failwith "difference between idents and trivia"
-        else
+        if lid.Length = trivia.Length then
+            List.zip lid trivia |> List.map SynIdent
+        elif lid.Length > trivia.Length then
+            let delta = lid.Length - trivia.Length
+            let trivia = [ yield! trivia; yield! List.replicate delta None ]
             List.zip lid trivia |> List.map SynIdent
+        else
+            failwith "difference between idents and trivia"
```
Comment (nojaf) [inline src/Compiler/SyntaxTree/SyntaxTree.fs:57]: @auduchinok what would you do in this case? 
This is just plain wrong if this state is ever achieved.

Comment (auduchinok) [reply src/Compiler/SyntaxTree/SyntaxTree.fs:57]: We can probably leave it as is if we aim to fix all places where these nodes are created.

File: src/Compiler/SyntaxTree/SyntaxTreeOps.fs [hunk 2]
```diff
@@ -468,7 +468,7 @@ let mkSynDotMissing mDot m l =
     | SynExpr.LongIdent (isOpt, SynLongIdent (lid, dots, trivia), None, _) ->
         // REVIEW: MEMORY PERFORMANCE: This list operation is memory intensive (we create a lot of these list nodes)
         SynExpr.LongIdent(isOpt, SynLongIdent(lid, dots @ [ mDot ], trivia), None, m)
-    | SynExpr.Ident id -> SynExpr.LongIdent(false, SynLongIdent([ id ], [ mDot ], []), None, m)
+    | SynExpr.Ident id -> SynExpr.LongIdent(false, SynLongIdent([ id ], [ mDot ], [ None ]), None, m)
```
Comment (nojaf) [inline src/Compiler/SyntaxTree/SyntaxTreeOps.fs:471]: @dsyme could we have this fix as well, please?

**Review by dsyme — APPROVED** [review]

File: src/Compiler/SyntaxTree/SyntaxTreeOps.fs (hunk 2 above)
Comment (dsyme) [reply src/Compiler/SyntaxTree/SyntaxTreeOps.fs:471]: @nojaf so this is all ready?

Comment (nojaf) [reply src/Compiler/SyntaxTree/SyntaxTreeOps.fs:471]: Yup, this is the most significant fix.
And if we did miss another case, `IdentsWithTrivia` will not blow up if a trivia is missing.

**Review by auduchinok — APPROVED** [review]

**Comment by abonie** [comment]
I am getting an error running `build.cmd` on main and I believe this PR is to blame. This is the error:
```
tests\service\SyntaxTreeTests\SynIdentTests.fs(19,49): error FS3174: The union case 'ParsedImplFileInput' does not have a field named 'modules'.
```
@nojaf

**Comment by nojaf** [comment]
https://github.com/dotnet/fsharp/commit/ff0f89217c9d99f80d5fbecad49dc0409e2a7e68

**Pull request merged**